airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -176,20 +179,32 @@ def _read_file(
             # check whether unstructured library is actually available for better error message and to ensure proper typing (can't be None after this point)
             raise Exception("unstructured library is not available")
 
-        filetype = self._get_filetype(file_handle, remote_file)
+        filetype: FileType | None = self._get_filetype(file_handle, remote_file)
 
-        if filetype == FileType.MD or filetype == FileType.TXT:
+        if filetype is None or filetype not in self._supported_file_types():
+            raise self._create_parse_error(
+                remote_file,
+                self._get_file_type_error_message(filetype),
+            )
+        if filetype in {FileType.MD, filetype is FileType.TXT}:
```

Comment:
I think this fixes the problem parsing records for unstrcutured.

```suggestion
        if filetype in {FileType.MD, FileType.TXT}:
```

Reply:
Oh, other tests are failing that we didn't know as we fail fast; interesting, checking now.

![image](https://github.com/user-attachments/assets/2ecd7928-8c10-408c-ae8c-46ba416f2071)

Reply:
[Checking here](https://github.com/airbytehq/airbyte-python-cdk/pull/51)